src/nsidc/metgen/metgen.py

```diff
@@ -255,7 +231,15 @@ def process(configuration: config.Config) -> None:
     """
     Process all Granules and record the results and summary.
     """
-    # TODO: Do any prep actions, like mkdir, etc
+    # TODO:
+    #   Do any prep actions, like mkdir, etc
+    #   Get real collection information from CMR
+    #   Determine data reader based on actual data file characteristics (e.g. suffix)
+    configuration.collection = Collection(
```

Comment:
`configuration` is already part of the setup for the partial setup (below), so adding the collection to the configuration seems logical given the current code. Is there a better way of splitting out this collection information without a huge refactor?

Reply:
I'm not sure I like it as part of the configuration. I'd like to keep that focused on what is read from the ini file and whatever commandline overrides and defaults are in place.

One thought is to keep it as-is, with the addition of a Python-y annotation to the `granule_collection` function, below.

Reply:
Oh, perhaps you're suggesting amended the `partial` step to add the Collection object?